accelerator/http.h

```diff
@@ -0,0 +1,53 @@
+#ifndef ACCELERATOR_HTTP_H_
+#define ACCELERATOR_HTTP_H_
+
+#include <microhttpd.h>
+#include <stdbool.h>
+#include "accelerator/apis.h"
+#include "accelerator/config.h"
+#include "accelerator/errors.h"
+#include "utils/logger_helper.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct ta_http_s {
+  bool running;
+  struct MHD_Daemon *daemon;
+  ta_core_t *core;
+} ta_http_t;
+
+/**
+ * Initializes an HTTP API
+ *
+ * @param http The HTTP status object
+ * @param core An TA config information
+ *
+ * @return a status code
+ */
+status_t ta_http_init(ta_http_t *const http, ta_core_t *const core);
+
+/**
+ * Starts an HTTP API
+ *
+ * @param api The HTTP API
+ *
+ * @return a status code
+ */
+status_t ta_http_start(ta_http_t *const http);
+
+/**
+ * Stops an HTTP API
+ *
+ * @param api The HTTP API
+ *
+ * @return a status code
+ */
+status_t ta_http_stop(ta_http_t *const http);
+
```

Comment:
Do we need a `destroy` function?

Reply:
I was intended to wrap microhttpd API calls into our function in `http.hc`.
This function would only be called when some terminated signals are triggered.

Reply:
I see. Also, if you intend to wrap like this, include of microhttpd should be in c file instead.